pkg/testing/fixture_install.go

```diff
@@ -302,6 +324,80 @@ func getElasticAgentProcesses(t *gotesting.T) []runningProcess {
 	return processes
 }
 
+// installDeb installs the prepared Elastic Agent binary from the deb
+// package and registers a t.Cleanup function to uninstall the agent if
+// it hasn't been uninstalled. It also takes care of collecting a
+// diagnostics when AGENT_COLLECT_DIAG=true or the test has failed.
+// It returns:
+//   - the combined output of Install command stdout and stderr
+//   - an error if any.
+func (f *Fixture) installDeb(ctx context.Context, installOpts *InstallOpts, opts []process.CmdOption) ([]byte, error) {
+	f.t.Logf("[test %s] Inside fixture installDeb function", f.t.Name())
+	//Prepare so that the f.srcPackage string is populated
+	err := f.EnsurePrepared(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to prepare: %w", err)
+	}
+
+	// sudo apt install the deb
+	out, err := exec.CommandContext(ctx, "sudo", "apt", "install", f.srcPackage).CombinedOutput() // #nosec G204 -- Need to pass in name of package
+	if err != nil {
+		return out, fmt.Errorf("apt install failed: %w output:%s", err, string(out))
+	}
+
+	f.t.Cleanup(func() {
+		f.t.Logf("[test %s] Inside fixture installDeb cleanup function", f.t.Name())
+		uninstallCtx, uninstallCancel := context.WithTimeout(context.Background(), 5*time.Minute)
+		defer uninstallCancel()
+		// stop elastic-agent, non fatal if error, might have been stopped before this.
+		f.t.Logf("running 'sudo systemctl stop elastic-agent'")
+		out, err := exec.CommandContext(uninstallCtx, "sudo", "systemctl", "stop", "elastic-agent").CombinedOutput()
+		if err != nil {
+			f.t.Logf("error systemctl stop elastic-agent: %s, output: %s", err, string(out))
+		}
+		// apt-get purge elastic-agent
+		f.t.Logf("running 'sudo apt-get -y -q purge elastic-agent'")
+		out, err = exec.CommandContext(uninstallCtx, "sudo", "apt-get", "-y", "-q", "purge", "elastic-agent").CombinedOutput()
+		if err != nil {
+			f.t.Logf("failed to apt-get purge elastic-agent: %s, output: %s", err, string(out))
+			f.t.FailNow()
+		}
+	})
+
+	// start elastic-agent
+	out, err = exec.CommandContext(ctx, "sudo", "systemctl", "start", "elastic-agent").CombinedOutput()
+	if err != nil {
+		return out, fmt.Errorf("systemctl start elastic-agent failed: %w", err)
+	}
+
+	// apt install doesn't enroll, so need to do that
+	enrollArgs := []string{"elastic-agent", "enroll"}
```

Comment:
Was this affected by https://github.com/elastic/elastic-agent/issues/4084? Or are we just logging errors and not actually changing the exit code of the command?

Reply:
I am interested to know why this doesn't doesn't catch that problem.

Reply:
Not sure I'm following.  If you run `dpkg -i elastic-agent-8.13.0-SNAPSHOT-arm64.deb`, to "install" the deb package, there is no place to add the enroll token.  So that is what is happening here.

Reply:
Right you need to run `elastic-agent enroll`. The "bug" in https://github.com/elastic/elastic-agent/issues/4084 is that if you do this right after `dpkg -i` it fails with an error because it can't restart the agent, because the service isn't running yet.

I now see you don't hit this problem because you `systemctl start elastic-agent` first. This is not what our default instructions for installing the deb suggest however, although it is a totally reasonable thing to do.

Reply:
I think we should add `systemctl start elastic-agent` to the DEB postinst script.  Any reason not to do that?

Reply:
I think it would prevent having `elastic-agent enroll --delay-enroll` work reliably now that I think about it a bit.

The use case would be to install the deb in a VM image and set it to enrol when it next boots up.

Reply:
I think this requires it's own issue to discuss I added #4381 to do that.  I think we can just leave this as-is for this PR, and adjust when we get #4381 figured out.